Multisig/Data/Services/Utils/HTTPClient.swift

```diff
@@ -4,61 +4,95 @@
         case unknownError(URLRequest, URLResponse?, Data?)
 
-        public var errorDescription: String? {
+        enum Message: String {
+            case networkRequestFailed = "No internet connection."
+            case entityNotFound = "Entity not found."
+            case invalidSafeChecksum = "Safe address checksum is not valid."
+            case unexpectedError = "Unexpected error. We are notified and will try to fix it asap."
+
+            /// Create a proper message from our backend internal code.
+            /// - Parameter code: backend internal code of error.
+            init(code: Int) {
+                #warning("TODO: this is a placeholder. Need to use real codes.")
+                switch code {
+                case 1000:
+                    self = .invalidSafeChecksum
+                default:
+                    #warning("TODO: Crashlytics.")
+                    self = .unexpectedError
+                }
+            }
+        }
+
+        var errorDescription: String? {
             switch self {
             case .networkRequestFailed(_, _, _):
-                return "The network request failed. Please try out later."
+                return Message.networkRequestFailed.rawValue
             case .entityNotFound(_, _, _):
-                return "Entity not found"
-            case .unprocessableEntity(_, _, _):
-                return "Unprocessable entity"
+                return Message.entityNotFound.rawValue
+            case .unprocessableEntity(_, _, let data):
+                guard let data = data else {
+                    #warning("TODO: Crashlytics")
+                    return Message.unexpectedError.rawValue
+                }
+                do {
+                    let details = try JSONDecoder().decode(ErrorDetails.self, from: data)
+                    guard let code = details.code else {
+                        #warning("TODO: Crashlytics")
+                        return Message.unexpectedError.rawValue
+                    }
+                    return Message(code: code).rawValue
+                } catch {
+                    #warning("TODO: Crashlytics")
+                    return Message.unexpectedError.rawValue
+                }
             default:
-                return "Unknown error"
+                #warning("TODO: Crashlytics")
+                return Message.unexpectedError.rawValue
             }
         }
-    }
-
-    private typealias URLDataTaskResult = (data: Data?, response: URLResponse?, error: Swift.Error?)
-
-    private let baseURL: URL
-    private let logger: Logger?
 
-    /// Creates new client with baseURL and logger
-    ///
-    /// - Parameters:
-    ///   - url: base url for creating all request urls
-    ///   - logger: logger for debugging and error purposes
-    public init(url: URL, logger: Logger? = nil) {
-        baseURL = url
-        self.logger = logger
+        struct ErrorDetails: Decodable {
+            let code: Int?
```

Comment:
Is there any need for code to be optional?

Reply:
you are right, no need in optional